llvm/test/CodeGen/DirectX/clean-module-flags2.ll

```diff
@@ -0,0 +1,17 @@
+; RUN: opt -S -dxil-prepare < %s | FileCheck %s
```

Comment:
It looks like `clean-module-flags.ll` and `clean-module-flags2.ll` are testing the same thing?  Or am I missing something?

Reply:
clean-module-flags.ll has Dwarf Version and Debug Info Version.

Reply:
Seems overkill to me - and by this precedent we'd need to add new combinations of flags is new ones ever show up. But probably doesn't hurt.

Reply:
Can we give this new test a more descriptive name? This is actually testing updating the flag behavior field, so how about "update-module-flag-behavior.ll"?

Reply:
Updated to legalize module flag.